lib/TBDGen/TBDGen.cpp

```diff
@@ -83,36 +99,35 @@ class TBDGenVisitor : public ASTVisitor<TBDGenVisitor> {
   void visitNominalTypeDecl(NominalTypeDecl *NTD) {
     auto declaredType = NTD->getDeclaredType()->getCanonicalType();
 
-    auto ntDescriptor = irgen::LinkEntity::forNominalTypeDescriptor(NTD);
-    addSymbol(ntDescriptor.mangleAsString());
+    addSymbol(irgen::LinkEntity::forNominalTypeDescriptor(NTD));
 
-    auto tmd = irgen::LinkEntity::forTypeMetadata(
+    addSymbol(irgen::LinkEntity::forTypeMetadata(
         declaredType, irgen::TypeMetadataAddress::AddressPoint,
-        /*isPattern=*/false);
-    addSymbol(tmd.mangleAsString());
-    auto tmda = irgen::LinkEntity::forTypeMetadataAccessFunction(declaredType);
-    addSymbol(tmda.mangleAsString());
-
-    if (isPrivateDecl(NTD))
-      return;
+        /*isPattern=*/false));
+    addSymbol(irgen::LinkEntity::forTypeMetadataAccessFunction(declaredType));
+
+    // There are symbols associated with any protocols this type conforms to.
+    for (auto conformance : NTD->getAllConformances()) {
```

Comment:
Isn't there a getLocalConformances() that only returns conformances defined in the same module?